pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -1585,18 +1589,35 @@ func (r *ReconcileInfinispan) serviceExternal(m *infinispanv1.Infinispan) *corev
 			Namespace: m.ObjectMeta.Namespace,
 			Labels:    lsService,
 		},
-		Spec: corev1.ServiceSpec{
+	}
+
+	if externalServiceType == "LoadBalancer" {
+		externalService.Spec = corev1.ServiceSpec{
+			Type:     externalServiceType,
+			Selector: lsPodSelector,
+		}
+	} else if externalServiceType == "NodePort" {
+		var portNum int32
+		if m.Spec.Expose.Ports != nil || len(m.Spec.Expose.Ports) > 0 {
+			portNum = m.Spec.Expose.Ports[0].NodePort
+		} else {
+			portNum = 0
+		}
+		externalService.Spec = corev1.ServiceSpec{
```

Comment:
Do you think to change Expose  from v1.ServiceSpec to the custom one, as we are using just two parameters, but customer need to create a complex one for real system. Or another parameters will be using in the fuster?

Reply:
I would prefer to use standard types, there are other kind of services like "External Name" (?) that maybe we want to support.

Reply:
Yes, sure. If we will cut to custom types, we will restrict it in the future. Let's keep the current implementation.